thirdparty/github.go

```diff
@@ -336,15 +340,33 @@ func getGithubClient(token, caller string, config retryConfig) *evergreen.GitHub
 // installation token is valid for a single GitHub API request.
 const defaultGitHubAPIRequestLifetime = 15 * time.Minute
 
-// getInstallationToken creates an installation token using Github app auth.
+// getCachedInstallationToken creates an installation token using Github app auth.
+// If possible, it will try to use an existing installation token for the app from the cache.
+// This should be used for all internal evergreen github operations.
 // If creating a token fails it will return the legacyToken.
-func getInstallationToken(ctx context.Context, owner, repo string, opts *github.InstallationTokenOptions) (string, error) {
+func getCachedInstallationToken(ctx context.Context, owner, repo string, opts *github.InstallationTokenOptions) (string, error) {
+	return installationToken(ctx, owner, repo, opts, true)
+}
+
+// GetInstallationToken creates an installation token using Github app auth.
+// It will not try to use the cache. This should only be used if there is a reason why a
+// cached token should not be used. If creating a token fails it will return the legacyToken.
+func GetInstallationToken(ctx context.Context, owner, repo string, opts *github.InstallationTokenOptions) (string, error) {
+	return installationToken(ctx, owner, repo, opts, false)
+}
+
+func installationToken(ctx context.Context, owner, repo string, opts *github.InstallationTokenOptions, useCached bool) (string, error) {
```

Comment:
I'm okay with either setup. There's also an argument to be made that it should always default to the cached token unless there is a specific reason why a non cached token is needed, which I believe will be rare. I can also solve this by adding more comments. 
@Kimchelly I'll leave it up to you.

Reply:
I believe the decision to use a cached or uncached token has to be made deliberately. Caching is meant to be a workaround to the problem where some of Evergreen's most common operations (e.g. getting GitHub API info, or getting a token for git.get_project cloning)  where we know that users can't directly interact with the token except in a specific/highly controlled way. If the operation rarely happens or somehow a user could directly get the token, then it's better to give them a fresh (no cache) token because Evergreen needs to reuse that token for its own internal purposes.

Since fetching artifact data is a pretty infrequent operation (I'd be surprised if more than a couple of hosts per day are spawned with task data) and users could get the token by querying Jasper, I'd say that `evergreen fetch` should not use a cached token at all.